### PR TITLE
MGMT-2170 - fix the mapping between machine and BMH

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**

### CODE REVIEW

File: internal/installcfg/installcfg.go [hunk 1]
```diff
@@ -180,6 +180,9 @@ func setBMPlatformInstallconfig(log logrus.FieldLogger, cluster *common.Cluster,
 	sortedHosts := make([]*models.Host, len(cluster.Hosts))
 	copy(sortedHosts, cluster.Hosts)
 	sort.Slice(sortedHosts, func(i, j int) bool {
+		if sortedHosts[i].Role != sortedHosts[j].Role {
```
Comment (eranco74) [inline internal/installcfg/installcfg.go:183]: Why this logic?

Comment (yevgeny-shnaidman) [reply internal/installcfg/installcfg.go:183]: so that the masters will be the first hosts defined, before workers

Comment (eranco74) [reply internal/installcfg/installcfg.go:183]: Consider adding a comment about the logic:
// Sorting hosts by by hostname, but master takes precedence before worker.

**Pull request merged**